docs/messaging/gcp-pubsub.md

```diff
@@ -19,4 +19,43 @@ For Google Cloud Pub/Sub, the following additional attributes are defined:
 | [`messaging.gcp_pubsub.message.ordering_key`](../attributes-registry/messaging.md) | string | The ordering key for a given message. If the attribute is not present, the message does not have an ordering key. | `ordering_key` | Conditionally Required: If the message type has an ordering key set. |
 <!-- endsemconv -->
 
+## Examples
+
+### Asynchronous Batch Publish Example
+
+Given is a process P that asynchronously publishes 2 messages in a batch to a topic T on Pub/Sub.
+
+```mermaid
+flowchart LR;
+  subgraph PRODUCER
+  direction LR
+  CA[Span Create A]
+  CB[Span Create B]
+  P[Span Publish A B] 
+  end
+  CA-. link .-P;
+  CB-. link .-P;
+
+  classDef producer fill:green
+  class P,CA,CB producer
+  classDef normal fill:green
+  class PA,PB,D1 normal
+  linkStyle 0,1 color:green,stroke:green
+```
+
+| Field or Attribute | Span Create A | Span Create B | Span Publish A B |
+|-|-|-|-|
+| Span name | `T create` | `T create` | `publish` |
+| Parent |  |  |  |
+| Links |  |  | Span Create A, Span Create B |
+| SpanKind | `PRODUCER` | `PRODUCER` | `PRODUCER` |
+| Status | `Ok` | `Ok` | `Ok` |
+| `messaging.batch.message_count` |  |  | 2 |
+| `messaging.destination.name` | `"T"` | `"T"` |  |
+| `messaging.destination.template` | `"topic"` | `"topic"` |  |
```

Comment:
```suggestion
```

the template is used when destination name has high cardinality (e.g. when topics are created dynamically and short-lived) and template would look like HTTP route (`/topic{customer}`.

We also no longer report  the kind`(topic|queue`) - https://github.com/open-telemetry/opentelemetry-specification/issues/3326